Cannot send a message to our peer when we are not connected!

**Describe the bug**
I am starting my neutrino node on  8b85751b461d2ec4151f12265ae4fb2cf9fe40c9 and I run into this error message: 

### First attempt at running
>./bitcoin-s-server
2020-07-10T13:41:53 INFO [HTTP] Initialized chain api
2020-07-10T13:41:54 INFO [WALLET] Using pre-existing wallet
2020-07-10T13:41:55 INFO [P2P] Sending getdata=GetDataMessage(inventoryCount=1, inventories=Vector(Inventory(MsgBlock,DoubleSha256Digest(b639ca945ef5833f8254988969906555654d27da080015000724210...) to peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:41:55 DEBUG [P2P] Sending msg=getdata to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:55 ERROR [P2P] Cannot send a message to our peer when we are not connected! payload=GetDataMessage(inventoryCount=1, inventories=Vector(Inventory(MsgBlock,DoubleSha256Digest(b639ca945ef5833f8254988969906555654d27da080015000724210...) peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:41:55 INFO [P2P] Starting node
2020-07-10T13:41:57 INFO [P2P] Attempting to connect to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:57 DEBUG [P2P] Tcp connection to: neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:57 DEBUG [P2P] Local: /192.168.1.187:45212
2020-07-10T13:41:57 INFO [P2P] Connection established with peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:41:57 DEBUG [P2P] Sending msg=version to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:57 INFO [P2P] We've been disconnected by Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) command=PeerClosed
[ERROR] [07/10/2020 08:41:57.300] [bitcoin-s-akka.actor.internal-dispatcher-3] [akka://bitcoin-s/user/P2PClient$-1594388515887] Cannot disconnect from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) when in state=Initializing
java.lang.RuntimeException: Cannot disconnect from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) when in state=Initializing
        at org.bitcoins.node.networking.peer.PeerMessageReceiver.disconnect(PeerMessageReceiver.scala:82)
        at org.bitcoins.node.networking.P2PClientActor.org$bitcoins$node$networking$P2PClientActor$$handleEvent(P2PClient.scala:175)
        at org.bitcoins.node.networking.P2PClientActor.org$bitcoins$node$networking$P2PClientActor$$handleTcpMessage(P2PClient.scala:132)
        at org.bitcoins.node.networking.P2PClientActor$$anonfun$org$bitcoins$node$networking$P2PClientActor$$awaitNetworkRequest$1.applyOrElse(P2PClient.scala:92)
        at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:35)
        at akka.event.LoggingReceive.apply(LoggingReceive.scala:96)
        at akka.event.LoggingReceive.apply(LoggingReceive.scala:70)
        at scala.PartialFunction.applyOrElse(PartialFunction.scala:187)
        at scala.PartialFunction.applyOrElse$(PartialFunction.scala:186)
        at akka.event.LoggingReceive.applyOrElse(LoggingReceive.scala:70)
        at akka.actor.Actor.aroundReceive(Actor.scala:535)
        at akka.actor.Actor.aroundReceive$(Actor.scala:533)
        at org.bitcoins.node.networking.P2PClientActor.aroundReceive(P2PClient.scala:53)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:577)
        at akka.actor.ActorCell.invoke(ActorCell.scala:547)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:270)
        at akka.dispatch.Mailbox.run(Mailbox.scala:231)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:243)
        at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
        at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
        at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
        at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)


From what I can tell, it seems that the bitcoind neutrino node is disconnecting us for some reason, which is waht I think this sequence of logs is indicating 

>2020-07-10T13:41:55 INFO [P2P] Starting node
2020-07-10T13:41:57 INFO [P2P] Attempting to connect to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:57 DEBUG [P2P] Tcp connection to: neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:57 DEBUG [P2P] Local: /192.168.1.187:45212
2020-07-10T13:41:57 INFO [P2P] Connection established with peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:41:57 DEBUG [P2P] Sending msg=version to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:41:57 INFO [P2P] We've been disconnected by Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) command=PeerClosed



### Second attempt at running

This produced different error messages, here they are. Notably it doesn't seem taht we got timeout problems: 

>2020-07-10T13:45:45 INFO [HTTP] Initialized chain api
2020-07-10T13:45:45 INFO [WALLET] Using pre-existing wallet
2020-07-10T13:45:47 INFO [P2P] Sending getdata=GetDataMessage(inventoryCount=1, inventories=Vector(Inventory(MsgBlock,DoubleSha256Digest(b639ca945ef5833f8254988969906555654d27da080015000724210...) to peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:45:47 DEBUG [P2P] Sending msg=getdata to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:47 ERROR [P2P] Cannot send a message to our peer when we are not connected! payload=GetDataMessage(inventoryCount=1, inventories=Vector(Inventory(MsgBlock,DoubleSha256Digest(b639ca945ef5833f8254988969906555654d27da080015000724210...) peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:45:47 INFO [P2P] Starting node
2020-07-10T13:45:48 INFO [P2P] Attempting to connect to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:48 DEBUG [P2P] Tcp connection to: neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:48 DEBUG [P2P] Local: /192.168.1.187:45750
2020-07-10T13:45:48 INFO [P2P] Connection established with peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)
2020-07-10T13:45:48 DEBUG [P2P] Sending msg=version to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:48 DEBUG [P2P] Received 151 TCP bytes
2020-07-10T13:45:48 DEBUG [P2P] Parsed 2 message(s) from bytes: NetworkMessage(NetworkHeader(TestNet3,version,UInt32Impl(103),ByteVector(4 bytes, 0xe0b3d712)), VersionMessage(ProtocolVersion70015, ServiceIdentifier(network=true, compactFilters=true, getUtxo=false, bloom=false, witness=true, xthin=false, networkLimited=true), epoch=1594388748, receiverServices=/142.147.59.88, receiverAddress=/142.147.59.88, receiverPort=45750), userAgent=/Satoshi:0.18.99/, startHeight=1780338, relay=true)), NetworkMessage(NetworkHeader(TestNet3,verack,UInt32Impl(0),ByteVector(4 bytes, 0x5df6e0e2)), VerAckMessage)
2020-07-10T13:45:48 DEBUG [P2P] Unaligned bytes after this: 0
2020-07-10T13:45:48 DEBUG [P2P] Received message=version from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Initializing
2020-07-10T13:45:48 DEBUG [P2P] Sending msg=verack to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:48 DEBUG [P2P] Received message=verack from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Initializing
2020-07-10T13:45:48 INFO [P2P] Our peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) has been initialized
2020-07-10T13:45:48 INFO [P2P] Our node has been full started. It took=1432ms
2020-07-10T13:45:48 DEBUG [chain-verification] Querying for best block hash
2020-07-10T13:45:49 DEBUG [P2P] Received message=sendheaders from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal
2020-07-10T13:45:49 DEBUG [P2P] Sending msg=sendheaders to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:49 DEBUG [P2P] Received message=ping from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal
2020-07-10T13:45:49 DEBUG [P2P] Sending msg=pong to peer=neutrino.testnet3.suredbits.com/34.210.154.40:18333
2020-07-10T13:45:49 DEBUG [P2P] Received message=addr from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal
2020-07-10T13:45:49 DEBUG [P2P] Received message=feefilter from peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) state=Normal
2020-07-10T13:45:49 INFO [P2P] We've been disconnected by Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333) command=PeerClosed
2020-07-10T13:45:49 DEBUG [P2P] Disconnected bitcoin peer=Peer(neutrino.testnet3.suredbits.com/34.210.154.40:18333)

--- a/app/server/src/main/scala/org/bitcoins/server/Main.scala
+++ b/app/server/src/main/scala/org/bitcoins/server/Main.scala
@@ -84,7 +84,7 @@ object Main extends App with BitcoinSLogger {
     }
 
     //get our wallet
-    val walletF = for {
+    val configuredWalletF = for {
       _ <- configInitializedF
       uninitializedNode <- uninitializedNodeF
       chainApi <- chainApiF
@@ -93,30 +93,29 @@ object Main extends App with BitcoinSLogger {
                                         chainApi,
                                         BitcoinerLiveFeeRateProvider(60),
                                         bip39PasswordOpt)
-      _ = logger.info("Starting wallet")
-      _ <- wallet.start()
-    } yield wallet
+    } yield {
+      logger.info(s"Done configuring wallet")
+      wallet
+    }
 
     //add callbacks to our uninitialized node
-    val nodeWithCallbacksF = for {
+    val configuredNodeF = for {
       uninitializedNode <- uninitializedNodeF
-      wallet <- walletF
+      wallet <- configuredWalletF
       initNode <- addCallbacksAndBloomFilterToNode(uninitializedNode, wallet)
-    } yield initNode
-
-    //start and sync our node
-    val startedNodeF = for {
-      node <- nodeWithCallbacksF
-      _ <- node.start()
-    } yield node
+    } yield {
+      logger.info(s"Done configuring node")
+      initNode
+    }
 
     //start our http server now that we are synced
     val startFut = for {
-      node <- startedNodeF
-      wallet <- walletF
+      node <- configuredNodeF
+      wallet <- configuredWalletF
+      _ <- node.start()
+      _ <- wallet.start()
       binding <- startHttpServer(node, wallet, rpcPortOpt)
-
-      _ =
+      _ = {
         if (nodeConf.isSPVEnabled) {
           logger.info(s"Starting SPV node sync")
         } else if (nodeConf.isNeutrinoEnabled) {
@@ -124,6 +123,7 @@ object Main extends App with BitcoinSLogger {
         } else {
           logger.info(s"Starting unknown type of node sync")
         }
+      }
       _ = BitcoinSServer.startedFP.success(Future.successful(binding))
 
       _ <- node.sync()

